File: backend/app/schemas/notification.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationRead(BaseModel):
    id: int
    user_id: int
    workspace_id: int | None = None
    type: str
    title: str
    message: str
    meeting_id: int | None = None
    read_at: datetime | None = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class NotificationListResponse(BaseModel):
    items: list[NotificationRead]
    total: int
    unread_count: int
    page: int
    size: int


class UnreadCountResponse(BaseModel):
    unread_count: int

File: frontend/src/types/notification.ts
export interface NotificationRead {
  id: number;
  user_id: number;
  workspace_id?: number | null;
  type: string;
  title: string;
  message: string;
  meeting_id?: number | null;
  read_at?: string | null;
  created_at: string;
}

export interface NotificationListResponse {
  items: NotificationRead[];
  total: number;
  unread_count: number;
  page: number;
  size: number;
}

export interface UnreadCountResponse {
  unread_count: number;
}
